packages/google_sign_in/google_sign_in_platform_interface/lib/src/types.dart

```diff
@@ -22,6 +22,55 @@ enum SignInOption {
   games
 }
 
+/// The parameters to use when initializing the sign in process.
+///
+/// The [hostedDomain] argument specifies a hosted domain restriction. By
+/// setting this, sign in will be restricted to accounts of the user in the
+/// specified domain. By default, the list of accounts will not be restricted.
+///
+/// The list of [scopes] are OAuth scope codes to request when signing in.
+/// These scope codes will determine the level of data access that is granted
+/// to your application by the user. The full list of available scopes can be
+/// found here: <https://developers.google.com/identity/protocols/googlescopes>
+///
+/// The [signInOption] determines the user experience. [SigninOption.games] is
+/// only supported on Android.
+///
+/// The [forceCodeForRefreshToken] is used on Android to ensure the authentication
+/// code can be exchanged for a refresh token after the first request.
```

Comment:
No need to duplicate all of the field documentation in the class-level documentation.